### PR TITLE
Adding a Pulumi 3.0 migration guide

### PR DESCRIPTION
**Pull request by stack72**

### CODE REVIEW

File: content/docs/get-started/install/migrating-3.0.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+---
+title: Migrating to Pulumi 3.0
+meta_desc: This page provides instructions for migrating to Pulumi 3.0
+no_on_this_page: true
+---
+
+## Migrating to 3.0
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:7]:
```suggestion
## Upgrading to Pulumi 3.0
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+---
+title: Migrating to Pulumi 3.0
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:2]:
```suggestion
title: Upgrading to Pulumi 3.0
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+meta_desc: This page provides instructions for migrating to Pulumi 3.0
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:3]:
```suggestion
meta_desc: This page provides instructions for upgrading to Pulumi 3.0
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+no_on_this_page: true
+---
+
+## Migrating to 3.0
+
+Upgrading to 3.0 is simple. First, you will [install the 3.0 CLI]({{< relref "/docs/get-started/install" >}}). Then, update each of your Pulumi programs to utilize the new SDK.
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:9]:
```suggestion
Pulumi 3.0 is currently in beta. If you'd like to try the beta and provide feedback, it's simple to upgrade. First, [install the 3.0 CLI]({{< relref "/docs/get-started/install" >}}). Then, update each of your Pulumi programs to utilize the new SDK.
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+## Changes to the CLI Behaviour
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:11]:
```suggestion
## CLI behavior changes in Pulumi 3.0
```

Do we have a standard style guide/point of view on American vs. British spellings?

File: content/docs/get-started/install/migrating-3.0.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+Previously, when using the `--stack` option on CLI commands, Pulumi would inconsistenty set that stack as the current stack. This behaviour has now standardised and will *NOT* set the stack as the current stack. This affects:
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:13]:
```suggestion
Previously, when using the `--stack` option on CLI commands, Pulumi would inconsistently set that stack as the current stack. In Pulumi 3.0, this behavior has now standardized and will *NOT* set the stack as the current stack. This affects the following commands:
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+```csharp
+<PackageReference Include="Pulumi" Version="3.*" />
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+## Automation API
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:89]:
```suggestion
## Automation API changes in Pulumi 3.0
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+The Pulumi 3.0 launch, moves the "Automation API" out of preview to be GA! This means, we have standardized the namespace requirements for the "Automation API". You can update your
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:91]:
```suggestion
The Pulumi 3.0 beta includes the final preview of the Automation API. That preview standardizes the namespace requirements for Automation API. You can update your
```

File: content/docs/get-started/install/migrating-3.0.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+{{< chooser language "javascript,typescript,python,go,csharp" >}}
+
+{{% choosable language "javascript,typescript" %}}
+
+```javascript
+// before
+import as automation from "@pulumi/pulumi/automation";
+
+//after
+import as automation from "@pulumi/pulumi/x/automation";
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:103]: I think these are backwards?

File: content/docs/get-started/install/migrating-3.0.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+{{% choosable language "csharp" %}}
+
+```csharp
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Pulumi.Automation;
+```
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:151]: Was there a before here or did C# never use the old namespace?

File: content/docs/get-started/install/migrating-3.0.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+{{< /chooser >}}
+
+## Remaining on Pulumi 3.0
+
+We recommend switching to Pulumi 3.0 if possible. We will only push critical security and bug fixes into the `3.x` branch. Other fixes, feature enhancements, and
+new functionality will not be supported in the `3.x` branch. In addition, provider updates will only be built against Pulumi 3.0.
+
+If you wish to remain on the `3.x` CLI, you can continue to download the CLI by referring to [the manual installation instructions]({{< relref "/docs/get-started/install#manual-installation" >}}) and [choosing a specific version]({{< relref "/docs/get-started/install/versions" >}}).
+
+`pulumi new` will attempt to use the latest versions of the templates, which pull in the `3.0` SDK. You can continue to use the `2.x` templates by running `pulumi new https://github.com/pulumi/templates/tree/2.x`.
```
Comment (infin8x) [inline content/docs/get-started/install/migrating-3.0.md:164]:
```suggestion
## When you should upgrade to Pulumi 3.0

Pulumi 3.0 is currently in beta and not recommended for existing production workloads. Once it is made generally available (GA), we will recommend switching to Pulumi 3.0 in the docs and in the CLI via the normal upgrade prompts. 

`pulumi new` uses the latest versions of the templates, which pull in the `3.0` SDK. You can continue to use the `2.x` templates by running `pulumi new https://github.com/pulumi/templates/tree/2.x`.
```

**Review by infin8x — APPROVED** [review]
Thanks for getting this started! Left some comments/suggestions.

File: content/docs/get-started/install/migrating-3.0.md (hunk 5 above)
Comment (stack72) [reply content/docs/get-started/install/migrating-3.0.md:11]: American - I just speak the correct version of English ;) J/k!

File: content/docs/get-started/install/migrating-3.0.md (hunk 9 above)
Comment (stack72) [reply content/docs/get-started/install/migrating-3.0.md:103]: great catch!

File: content/docs/get-started/install/migrating-3.0.md (hunk 10 above)
Comment (stack72) [reply content/docs/get-started/install/migrating-3.0.md:151]: The DotNet stuff is only new and didn't follow the old X namespace work

**Pull request merged**